### ISSUE TITLE
regarding building image patches for training set construction

### ISSUE THREAD

**Issue by surfreta** [azavea/raster-vision#125]
Hi, Thanks for sharing the code.

I am working on a semantic segmentation problem over the high resolution aerial imagery. Instead of ISPRS data set, my studied problem has only two classes, class A and class B. Moreover, the proportion of class B is much smaller than class A. For instance, in a given image, maybe only 10% area belongs to class B. How should I construct the training patches for this kind of scenario?

**Comment by lewfish** [comment]
Sorry for the late reply. I think that you are referring to an imbalanced dataset, where some of the classes are represented far less often than others. Some people handle this by oversampling (feeding the rare classes into the learner extra often) or using a weighted loss function (which penalizes mistakes on the rare classes more often). I haven't had much luck with either of them myself. At some point we could add an oversampling option for `make_training_chips`, but it doesn't seem like a priority right now.

**Comment by Rabscuttler** [comment]
I am trying a similar kind of problem, finding solar PV arrays in high res aerial imagery. I would also welcome an option to address training on imbalanced classes, as it is not obvious to how to bolt on oversampling as is in chip creation, or indeed create custom loss functions with the existing backends.

**Comment by lewfish** [comment]
We have an option to do something like this in the fastai plugin for semantic segmentation: https://github.com/azavea/raster-vision-fastai-plugin/blob/master/fastai_plugin/semantic_segmentation_backend.py#L401-L407
